vars/job_monitoring.groovy

```diff
@@ -0,0 +1,75 @@
+import groovy.json.JsonSlurper
+
+/**
+ * Extracts the displayName from given build.
+ *
+ * @param response String Object containing the response from  previous HTTP GET request.
+ * @param upstream_status String Object indicates whether this is an upstream build.
+ */
+def build_info(response, upstream_status=null) {
+    def object = null
+    try {
+      def jsonSlurper = new JsonSlurper()
+      object = jsonSlurper.parseText(response)
+
+      if (upstream_status){
+        return object.description
+      }
+
+      def url = env.JENKINS_URL \
+             + object.actions.causes[0].upstreamUrl.join(", ") \
+             + object.actions.causes[0].upstreamBuild.join(", ") \
+             + "/api/json"
+
+      response = url.toURL().text
+      object = jsonSlurper.parseText(response)
+      } catch (Error e){}
```

Comment:
Catching Error is too generic. I don't really know groovy but the concept of exception handling is the same in every language, and the idea is that you should not catch too generic exceptions.
Please find the exception type that you need to catch and replace it with the generic Error.

Reply:
The catch line should be indented back to spaces to match the indentation of the try.
See example https://www.tutorialspoint.com/groovy/groovy_exception_handling.htm

Reply:
Ok

Reply:
After checking it more thoroughly, I came to decision to leave it as is.

Explanation: I agree with you that catching too broad exception is practice we should avoid. BUT, dealing with Jenkins pipelines written in Groovy, leads to unexpected behavior and time consuming debugging.

Catching board exceptions cover us from Groovy/Jenkins related unexpected behavior and pass all my tests so far.

So, with respect to all written above, I will leave it as is.

Reply:
What I mean by 'unexpected behavior' : sometimes the exception is thrown and sometimes not. When it is thrown, I don't understand its origin/cause. In addition, the types of these exceptions differ from time to time